extensions/wikia/Search/classes/Services/ESFandomSearchService.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace Wikia\Search\Services;
+
+class ESFandomSearchService extends ESEntitySearchService {
+	const RESULTS_COUNT = 6;
+
+	protected function prepareQuery( $phrase ) {
+		return urlencode( $phrase );
+	}
+
+	protected function select( $query ) {
+		global $wgConsulServiceTag, $wgConsulUrl;
+
+		$consulUrl =
+			( new \Wikia\Service\Gateway\ConsulUrlProvider( $wgConsulUrl,
+				$wgConsulServiceTag ) )->getUrl( 'fandom-search' );
+
+		$response = \Http::post( "http://$consulUrl/fandom", array(
+			'noProxy' => true,
+			'postData' => "queryTerms=$query",
+			'headers' => [
+				'Content-Type' => 'application/x-www-form-urlencoded',
+			],
+		) );
+		if ( $response !== false ) {
+
+			$decodedResponse = json_decode( $response, true );
+			if ( isset( $decodedResponse['hits']['hits'] ) &&
+			     json_last_error() === JSON_ERROR_NONE
+			) {
+				return $decodedResponse['hits']['hits'];
+			}
+		}
+
+		return [];
+	}
+
+	protected function consumeResponse( $response ) {
+		$results = [];
+
+		foreach ( $response as $item ) {
+			if ( isset( $item['_source'] ) ) {
+				$source = $item['_source'];
+
+				$results[] = [
+					'title' => html_entity_decode( $source['title'] ),
+					'excerpt' => isset( $source['excerpt'] )
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "excerpt" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)